contracts/Auction.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity ^0.8.0;
+
+contract Auction {
+    // current state of the auction
+    address payable public beneficiary;
+    uint256 public auctionEndTime;
+
+    address public highestBidder;
+    uint256 public highestbid;
+
+    enum Auc_State {
+        Running,
+        Ended,
+        Cancelled
+    }
+    Auc_State public auctionState;
+
+    mapping(address => uint256) pendingReturns;
+
+    event highestBidIncreased(address bidder, uint256 amount);
+    event auctionEnded(address winner, uint256 amount);
+
+    constructor(uint256 _biddingTime, address payable _beneficiary) {
+        auctionState = Auc_State.Running;
+        beneficiary = _beneficiary;
+        auctionEndTime = block.timestamp + _biddingTime;
+    }
+
+    modifier notOwner() {
+        require(msg.sender != beneficiary, "Owner cannot bid");
+        _;
+    }
+
+    modifier Owner() {
+        require(msg.sender == beneficiary, "Caller must be auctioner");
+        _;
+    }
+
+    modifier TimeLimit() {
+        require(block.timestamp < auctionEndTime, "The Auction Time Is Over");
+        _;
+    }
+
+    function bid() public payable notOwner TimeLimit {
+        require(
+            auctionState == Auc_State.Running,
+            "The Auction Is not Running"
+        );
+        if (msg.value > highestbid) {
+            if (pendingReturns[msg.sender] > 0) {
+                uint256 amount = pendingReturns[msg.sender];
+                payable(msg.sender).transfer(amount);
```

Comment:
You can use the call method instead of transfer which is the recommended way.
For more info: https://solidity-by-example.org/sending-ether/